pkg/operator/ceph/object/notification/obc_label_controller_test.go

```diff
@@ -263,6 +296,157 @@ func TestCephBucketNotificationOBCLabelController(t *testing.T) {
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 2, createNotificationCount)
+		assert.True(t, sets.NewString(createdNotifications...).Has(testNotificationName+"-1"))
+		assert.True(t, sets.NewString(createdNotifications...).Has(testNotificationName+"-2"))
+		assert.Equal(t, 0, deleteNotificationCount)
+	})
+
+	t.Run("delete multiple notifications from the obc", func(t *testing.T) {
+		resetValues()
+		multipleDeleteOBC, multipleDeleteOB := createOBResources(multipleDeleteBucketName)
+		multipleDeleteOBC.Spec.GenerateBucketName = multipleDeleteBucketName
+		multipleDeleteOBC.Spec.ObjectBucketName = multipleDeleteBucketName
+		multipleDeleteOBC.Status.Phase = bktv1alpha1.ObjectBucketClaimStatusPhaseBound
+		req.NamespacedName.Name = multipleDeleteBucketName
+		objects := []runtime.Object{
+			cephCluster,
+			multipleDeleteOBC,
+			multipleDeleteOB,
+		}
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileOBCLabels{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 0, createNotificationCount)
+		assert.Equal(t, 2, deleteNotificationCount)
+		assert.True(t, sets.NewString(deletedNotifications...).Has(multipleDeleteBucketName+testNotificationName+"-1"))
+		assert.True(t, sets.NewString(deletedNotifications...).Has(multipleDeleteBucketName+testNotificationName+"-2"))
+	})
+	t.Run("provision OBC with multiple delete and create of notifications", func(t *testing.T) {
+		resetValues()
+		multipleBothOBC, multipleBothOB := createOBResources(multipleBothBucketName)
+		multipleBothOBC.Labels = setNotificationLabels([]string{testNotificationName + "-1", testNotificationName + "-2"})
+		multipleBothOBC.Spec.ObjectBucketName = multipleBothBucketName
+		multipleBothOBC.Status.Phase = bktv1alpha1.ObjectBucketClaimStatusPhaseBound
+		bucketNotification1 := createBucketNotification(testNotificationName + "-1")
+		bucketNotification2 := createBucketNotification(testNotificationName + "-2")
+		req.NamespacedName.Name = multipleBothBucketName
+
+		objects := []runtime.Object{
+			cephCluster,
+			multipleBothOBC,
+			multipleBothOB,
+			bucketNotification1,
+			bucketNotification2,
+			bucketTopic,
+		}
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileOBCLabels{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 2, createNotificationCount)
+		assert.True(t, sets.NewString(createdNotifications...).Has(testNotificationName+"-1"))
+		assert.True(t, sets.NewString(createdNotifications...).Has(testNotificationName+"-2"))
```

Comment:
For the future, there is an assert builtin that outputs a bit better info when the test fails. `assert.ElementsMatch()`, and then we just have to build up a string slice of `[]string{testNotificationName+"-1", testNotificationName+"-2"}` With `ElementsMatch()`, we don't also need to check the count above either, since it will fail if more elements are present.